Docs: Add information on Environment Variables configuration (#717)
hyperledger/iroha: docs/source/configure/index.rst

  ``"initial_peers" : [{"address":"127.0.0.1:10001", "public_key":
  "bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308929"}]``

> Alternative

Reviewer: I would avoid marking this possibility as "alternative", which sounds like "secondary". The approach with environment is meant to provide completely the same functionality as the config file, so IMO they should be explained as equally adequate approaches. What about making two same-level sections, where one describes config file syntax, and the other is about env vars?

Author: Ok, will do!

===========

> Alternative to using a configuration file is setting up the configuration using environment variables.

Reviewer: strictly speaking, it is not an alternative, as both can be used at once, and your next sentence is just about that

> **Configuration file, if present, will have a priority in parameters that are set up in it.**

Reviewer: ```suggestion
**The parameters specified in the configuration file, if present, will override the ones that are set up through environment.**
```

> Here is how it will look like in

Reviewer: I would suggest to add a bit more formal description on how to set some parameter through env.
first, the lazy approach, let the user set the flag `--verbosity trace` and take the parameter name from the log.
second, the names are formed from the config structure: fixed path labels are taken as is, arrays are indexed starting from `0`, and user-provided dictionary keys are a bit trickier: the key and the value are set in separate vars. everything is uppercased and joined with `_`.

Unix
----

``export IROHA_BLOCK_STORE_PATH=/tmp/block_store/
export IROHA_TORII_PORT=50051
export IROHA_INTERNAL_PORT=10001
export IROHA_PG_OPT="host=172.19.0.2 port=5432 user=iroha password=helloworld"
export IROHA_MAX_PROPOSAL_SIZE=10
export IROHA_PROPOSAL_DELAY=5000
export IROHA_VOTE_DELAY=5000
export IROHA_MST_ENABLE=false
export IROHA_MST_EXPIRATION_TIME=1440
export IROHA_MAX_ROUNDS_DELAY=3000
export IROHA_CRYPTO_PROVIDERS_0_KEY=p1
export IROHA_CRYPTO_PROVIDERS_0_CRYPTO_TYPE=ed25519_sha3_256
export IROHA_CRYPTO_PROVIDERS_0_PRIVATE_KEY=cc5013e43918bd0e5c4d800416c88bed77892ff077929162bb03ead40a745e88
export IROHA_CRYPTO_PROVIDERS_0_TYPE=default
export IROHA_CRYPTO_SIGNER=p1

bin/irohad --genesis_block example/genesis.block --overwrite_ledger --verbosity=trace
``

Windows
-------

``setx IROHA_BLOCK_STORE_PATH=/tmp/block_store/
setx IROHA_TORII_PORT=50051
setx IROHA_INTERNAL_PORT=10001
.....
``

> Shell

Reviewer 2: ```suggestion
PowerShell
```

.....
``

Parameter names
---------------

As you can see, the parameter names are not the same as in the configuration file.

They are formed from the config structure, fixed label IROHA is added to the beginning and everything is uppercased and joined with _.
Let us look a bit closer at how they are structured:

**With simple string values**

In config file:

``"block_store_path" : "/tmp/block_store/" ``

In env variables:

``IROHA_BLOCK_STORE_PATH=/tmp/block_store/``

**With arrays**

Arrays are indexed starting with 0 and should be in direct order without skipping any numbers:

In config file:

``"initial_peers" : [{"address":"127.0.0.1:10001", "public_key": "bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308929"}, {"address":"127.0.0.1:10002", "public_key": "bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308920"}] ``

In env variables:

``IROHA_INITIAL_PEERS_0_ADDRESS=127.0.0.1:10001
IROHA_INITIAL_PEERS_0_PUBLIC_KEY=bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308929
IROHA_INITIAL_PEERS_1_ADDRESS=127.0.0.1:10002
IROHA_INITIAL_PEERS_1_PUBLIC_KEY=bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308920``

> **User-provided user dictionaries**

Reviewer: ```suggestion
**User-provided user dictionaries of users**
```

Author: Doesn't sound very nice. Can you please elaborate so I could probably try to make it look a bit better?...

Author: you're joking, right?)

Author: Should be better now :D

Author: Still waiting for @lebdron

Reviewer 2: Simple string values and arrays are also user-provided, so it looked like a joke. I think "user-provided" part can be removed, leaving just "dictionaries".

User-provided dictionary keys are a bit trickier: the key and the value are set in separate vars

In config file:

> ``"crypto": {"providers": {

Reviewer: maybe an example with loggers would be easier to grasp:

```
  "log": {
    "level": "info",
    "children": {
      "KeysManager": {
        "level": "trace"
      },
      "Irohad": {
        "level": "error"
      }
    }
  }
```

Reviewer 2: There is an example below

Reviewer: indeed

IROHA_INITIAL_PEERS_1_ADDRESS=127.0.0.1:10002
IROHA_INITIAL_PEERS_1_PUBLIC_KEY=bddd58404d1315e0eb27902c5d7c8eb0602c16238f005773df406bc191308920``

> **Dictionaries**

Reviewer: ```suggestion
**Dictionaries with user-defined keys**
```
